From 3cd2cce95c910446be970c93f9e7632a9c8a77a0 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 17 Nov 2019 13:43:19 +0200
Subject: [PATCH] [SW-6506] hl: exclude ppboot functionality from preboot

ppboot is doing all pcie craft and all needed to upload preboot

Change-Id: I153a9d7c701a9970a98ca643b26e8238d01b3b12
---

diff --git a/app/preboot.c b/app/preboot.c
index 330ac37..2abfb1a 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -96,18 +96,6 @@
 	regmap_writel(uboot_offset, mmPSOC_GLOBAL_CONF_SCRATCHPAD_30);
 }
 
-static void preboot_pcie_init(void)
-{
-	if (hl_skip_pcie_init()) {
-		preboot_puts("PCIe initialization is skipped!\n");
-		return;
-	}
-
-	//preboot_puts("Waiting for PCIe link...\n");
-	preboot_pcie_wait_for_link();
-	preboot_puts("PCIe link is UP!\n");
-}
-
 #ifndef TARGET_IS_TESTER
 static void preboot_version(void)
 {
@@ -153,28 +141,14 @@
 static inline void preboot_version(void) {}
 #endif	/* TARGET_IS_TESTER */
 
-static void hl_jtag_init(void)
-{
-	u32 reg, nPRESETDBG = BIT(12);
-
-	/* Enable JTAG attach */
-	reg = regmap_readl(mmCPU_CA53_CFG_ARM_RST_CONTROL);
-	reg |= nPRESETDBG;
-	regmap_writel(reg, mmCPU_CA53_CFG_ARM_RST_CONTROL);
-}
-
 void preboot_main(void)
 {
 	void (*spl_start)(void) = (void *)SPL_TEXT_BASE;
 
 	regmap_writel(CPU_BOOT_STATUS_IN_PREBOOT, get_cpu_boot_status_reg());
 
-	hl_jtag_init();
-	preboot_sync_btl();
 	hl_uart_init();
-	preboot_pcie_init();
 	preboot_version();
-	preboot_delay();
 	hl_pll_init();
 	hl_pll_wa();
 	/* reinit uart after PLLs configured */
diff --git a/gaudi/preboot.lds.S b/gaudi/preboot.lds.S
index 474bd2d..17a9257 100644
--- a/gaudi/preboot.lds.S
+++ b/gaudi/preboot.lds.S
@@ -5,7 +5,7 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-MEMORY { .sram : ORIGIN = 0x7FFBFE0000, LENGTH = 0x10000 }
+MEMORY { .sram : ORIGIN = 0x7FFBFE4000, LENGTH = 0x10000 }
 
 OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
 OUTPUT_ARCH(aarch64)
